backend/lib/pix-service/service.js

```diff
@@ -0,0 +1,41 @@
+import { Value } from '@sinclair/typebox/value'
+import { PIECE_INDEXER_URL } from '../config.js'
+import { getMinerPeerId } from '../rpc-service/service.js'
+import { PixResponse } from './data-types.js'
+import pRetry from 'p-retry'
+
+/**
+ * @param {string} providerId
+ * @param {string} pieceCid
+ * @returns {Promise<string>}
+ */
+export const pixRequest = async (providerId, pieceCid) => {
+  const url = PIECE_INDEXER_URL + '/sample/' + providerId + '/' + pieceCid
+  try {
+    const response = await pRetry(async () => await fetch(url, {
+      method: 'GET',
+      headers: { 'content-type': 'application/json' }
+    }), { retries: 5 })
+    const json = await response.json()
+    try {
+      const parsedPixResponse = Value.Parse(PixResponse, json)
+      return parsedPixResponse.samples[0]
+    } catch (e) {
+      e.message = `Failed to parse response from piece indexer: ${e.message}, response: ${JSON.stringify(json)}`
+      throw e
+    }
+  } catch (e) {
+    console.error(`Failed to make RPC request: ${e.message}`)
```

Comment:
```suggestion
    console.error('Failed to make RPC request', e)
```

If we only log `e.message`, we lose the stack trace. `Console.error` knows how to treat errors and log them accordingly:

```js
> console.log('hi', new Error('oh no'))
hi Error: oh no
    at REPL1:1:19
    at ContextifyScript.runInThisContext (node:vm:136:12)
    at REPLServer.defaultEval (node:repl:598:22)
    at bound (node:domain:432:15)
    at REPLServer.runBound [as eval] (node:domain:443:12)
    at REPLServer.onLine (node:repl:927:10)
    at REPLServer.emit (node:events:531:35)
    at REPLServer.emit (node:domain:488:12)
    at [_onLine] [as _onLine] (node:internal/readline/interface:417:12)
    at [_line] [as _line] (node:internal/readline/interface:888:18)
```